p2p/libp2p/identity/identity.go

```diff
@@ -0,0 +1,125 @@
+package identity
```

Comment:
I like where this is going but had to google what a multiaddr was. Were you thinking of adding tests for this in the future?

Reply:
I went back and forth on adding tests. After refactoring and looking at what's there, I got the impression we would either end up testing code that's already well tested (and just being composed) or would require extraneous effort to test (e.g. DNS resolution). Given the pace of iteration in this area and our current standing it seemed to me like the time was better spent elsewhere but would be more than happy to put more thought into a test plan if you feel like it would be prudent at this stage already.

Reply:
I hope to address any confusion around previously foreign concepts like multiaddr in forthcoming documentation updates.